System/Random.hs

```diff
@@ -201,6 +210,24 @@ class RandomGen g where
   split    :: g -> (g, g)
 
 
+-- | Default split function. This function should be safe to use as the 'split'
+-- function for most 'RandomGen' implementations.
+--
+-- It is recommended that you use a different 'split' implementation if the
+-- random number algorithm you are using has its own algorithm for 'split', or
+-- if you are implementing a cryptographic random number generator.
+defaultSplit :: RandomGen g => g -> (g, g)
+defaultSplit = go decoder where
+  decoder = runGetIncremental initialize
+
+  go :: RandomGen g => Decoder g -> g -> (g, g)
+  go (Done _leftover _consumed new) gen = (gen, new)
+  go (Partial k) gen = go (k (Just mixed)) gen' where
+    (bs, gen') = genByteString 8 gen
+    mixed = toStrict . bytestringDigest . sha256 $ fromStrict bs
+  go (Fail _leftover _consumed msg) _gen = error msg -- initialize should never fail
+  {-# INLINE go #-}
```

Comment:
`go` is a recursive function, there is no point on putting `INLINE` pragma on it. `defaultSplit` however could benefit from inlining.

Reply:
Thanks! Fixed.